packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx

```
@@ -157,25 +157,13 @@ const ContinuousBinningModal: React.FC<ContinuousBinningModalProps> = ({
     intervalForm.validate();
     // Adding form objects to dep array causes infinite rerenders
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [intervalForm.values]);
+  }, [intervalForm.getValues()]);
 
   useEffect(() => {
     if (binMethod === "interval") {
       rangeForm.clearErrors();
-      intervalForm.validate();
     } else {
       intervalForm.clearErrors();
-      rangeForm.validate();
```

This change is causing some incorrect behavior in the binning modal. If you switch modes, then you can save bins that don't follow our binning rules. Also, the last row is getting dropped? Not sure if that is due to this change or something else in the file:

https://github.com/NCI-GDC/gdc-frontend-framework/assets/4624053/771f35cb-4420-42cf-9a10-943f45ca3b36

---

will look into, it was not intentional to remove anything

---

pushed a fix, the problem was the `      rangeForm.clearFieldError(
        `ranges.${rangeForm.values.ranges.length - 1}.to`,
      );`
no longer worked as the validation was being done asynchronously 

but it looks to work fine with just putting the checks on the top part of the form as the bottom checks every time you add a value, let me know if this works for you

---

Hey @ocshawn, I'm still seeing the same issue if you switch to the range form when you have interval validation errors and then switch back to the interval form, you can save the bins when you shouldn't be able to. I'll look at the mantine form docs and see if there's a new way to replicate what these lines were doing 🤔

---

@amlehman thanks for looking, i think the whole form may need to be redone, or we could just remove this section of the code that removed the errors and let the errors stay in the grayed out area

---

Hey, @ocshawn. Yeah, I'm not finding an easy way to fix the behavior. I think what we do with it is @wteouchicago 's call. There were a ton of requirements for the way this form/validation behaves, not sure what is negotiable.

---

Hmm, so it's one that allowing users to save bins that make no sense, but another thing that we seem to be showing the wrong info after allowing them to save "bad" bins (see attachment -- that case count of 75 should belong to the bin named "20-30".

<non-technical hand-waving> can we do something like this: always revalidate all the fields when the user clicks Save Bins and don't save the bins if there are errors? Let's Slack on this if it gets complicated.

https://github.com/NCI-GDC/gdc-frontend-framework/assets/73256434/d1ea93e8-2d12-4400-945c-251c5a1bc213

---

i uploaded one potential fix, you can still see the error messages in the grayed out area, if this does not work i propose we hide the error messages with css when the area is not selected

---

@ocshawn this works great! I think it's fine that users can still see error messages in the greyed out area. 👍